File: api-asn/csharp/src/api/asn/esya/EESYASDOHash.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using tr.gov.tubitak.uekae.esya.api.asn;
using tr.gov.tubitak.uekae.esya.api.asn.esya;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.asn.esya;

namespace tr.gov.tubitak.uekae.esya.api.asn.esya
{
    public class EESYASDOHash : BaseASNWrapper<ESYASDOHash>,ExtensionType
    {
        public EESYASDOHash(ESYASDOHash aObject):base(aObject) {}

        public EESYASDOHash(byte[] aBytes) : base(aBytes, new ESYASDOHash()){}

        public EESYASDOHash( EAlgorithmIdentifier hashAlgorithm, byte[] hashValue) 
            :base(new ESYASDOHash(hashAlgorithm.getObject(), hashValue)){}

        public EAlgorithmIdentifier getHashAlgorithm(){
            return new EAlgorithmIdentifier(mObject.hashAlgorithm);
        }

        public byte[] getHashValue(){
            return mObject.hashValue.mValue;
        }

        public EExtension toExtension(bool aCritic){
            return new EExtension(EESYAOID.oid_ESYA_SDO, aCritic, getBytes());
        }
    }
}

File: api-asn/src/tr/gov/tubitak/uekae/esya/api/asn/esya/EESYASDOHash.java
package tr.gov.tubitak.uekae.esya.api.asn.esya;

import tr.gov.tubitak.uekae.esya.api.asn.BaseASNWrapper;
import tr.gov.tubitak.uekae.esya.api.asn.x509.EAlgorithmIdentifier;
import tr.gov.tubitak.uekae.esya.api.asn.x509.EExtension;
import tr.gov.tubitak.uekae.esya.api.asn.x509.ExtensionType;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.asn.esya.ESYASDOHash;

/**
 <b>Author</b>    : zeldal.ozdemir <br>
 <b>Project</b>   : MA3   <br>
 <b>Copyright</b> : TUBITAK Copyright (c) 2006-2011 <br>
 <b>Date</b>: 8/9/11 - 4:32 PM <p>
 <b>Description</b>: <br>
 */
public class EESYASDOHash extends BaseASNWrapper<ESYASDOHash> implements ExtensionType{
    public EESYASDOHash(ESYASDOHash aObject) {
        super(aObject);
    }

    public EESYASDOHash(byte[] aBytes) throws ESYAException {
        super(aBytes, new ESYASDOHash());
    }

    public EESYASDOHash( EAlgorithmIdentifier hashAlgorithm, byte[] hashValue) {
        super(new ESYASDOHash(hashAlgorithm.getObject(), hashValue));
    }

    public EAlgorithmIdentifier getHashAlgorithm(){
        return new EAlgorithmIdentifier(mObject.hashAlgorithm);
    }

    public byte[] getHashValue(){
        return mObject.hashValue.value;
    }

    public EExtension toExtension(boolean aCritic) throws ESYAException {
        return new EExtension(EESYAOID.oid_ESYA_SDO, aCritic, this);
    }
}
